presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -287,6 +295,8 @@
     private final AccessControl accessControl;
     private final WarningCollector warningCollector;
 
+    private QuerySpecification currentSubquery;
```

Comment:
This does not seem right to me. Why do we need a root level field?

Reply:
I think we should fix this, let me think about it.

Reply:
Can we check if we can use Analysis object to store this information?

Reply:
@jainxrohit I could add it directly as a field to `Analysis` which would be no problem. Did you have something more than that in mind?

Reply:
One way of doing it would be to rewrite all logic thats run when analyzeFrom is called again. We should only need some small parts which affect the scope